railgun/internal/diagnostics/server.go

```diff
@@ -46,8 +63,22 @@ func (s *Server) Listen(ctx context.Context) error {
 	}
 }
 
-// installHandlers adds the Profiling webservice to the given mux.
-func installHandlers(mux *http.ServeMux) {
+// receiveConfig watches the config update channel
+func (s *Server) receiveConfig(ctx context.Context) {
+	for {
+		select {
+		case dump := <-s.ConfigDumps.SuccessfulConfigs:
+			successfulConfigDump = dump
+		case dump := <-s.ConfigDumps.FailedConfigs:
+			failedConfigDump = dump
+		case <-ctx.Done():
+			return
```

Comment:
When dealing with a server running asynchronously as a goroutine within the greater program, it's always a good idea to do some logging and error checking on context completion to help operators understand better the state of the program:

```suggestion
		case <-ctx.Done():
		if err := ctx.Err(); err != nil {
		    // do something with the error, if not just logging
		}
		// log context completion is causing server shutdown
			return
```

Reply:
Worked into 98cacd13e4bfd8bfc9d3650fbf99ccd2437943eb